### ISSUE TITLE
Socket Exception when running tests with --explore option (3.6)

### ISSUE THREAD

**Issue by oryz** [nunit/nunit#2027]
Hi,

I'm not sure whether an issue with NUnit or with my configuration, but when trying to run tests with the `--explore` option, it errors with the following output:

```
Runtime Environment
   OS Version: Microsoft Windows NT 6.1.7601 Service Pack 1
  CLR Version: 4.0.30319.42000

Test Files
    C:\...\some.dll

Test Filters
    Where: cat=='cat_name'

System.Net.Sockets.SocketException (0x80004005): An existing connection was forcibly closed by the remote host

Server stack trace:
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size,
 SocketFlags socketFlags)
   at System.Runtime.Remoting.Channels.SocketStream.Read(Byte[] buffer, Int32 of
fset, Int32 size)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadFromSocket(Byte[] buffe
r, Int32 offset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.Read(Byte[] buffer, Int32 o
ffset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadAndMatchFourBytes(Byte[
] buffer)
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadAndMatchPreamble
()
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadVersionAndOperat
ion(UInt16& operation)
   at System.Runtime.Remoting.Channels.Tcp.TcpClientSocketHandler.ReadHeaders()
   at System.Runtime.Remoting.Channels.Tcp.TcpClientTransportSink.ProcessMessage
(IMessage msg, ITransportHeaders requestHeaders, Stream requestStream, ITranspor
tHeaders& responseHeaders, Stream& responseStream)
   at System.Runtime.Remoting.Channels.BinaryClientFormatterSink.SyncProcessMess
age(IMessage msg)

Exception rethrown at [0]:
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage req
Msg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgDa
ta, Int32 type)
   at NUnit.Engine.ITestAgent.Stop()
   at NUnit.Engine.Runners.ProcessRunner.Dispose(Boolean disposing)
   at NUnit.Engine.Runners.AbstractTestRunner.Dispose()
   at NUnit.Engine.Runners.MasterTestRunner.Dispose(Boolean disposing)
   at NUnit.Engine.Runners.MasterTestRunner.Dispose()
   at NUnit.ConsoleRunner.ConsoleRunner.ExploreTests(TestPackage package, TestFi
lter filter)
   at NUnit.ConsoleRunner.Program.Main(String[] args)
```

The command I executed was:
> "nunit3-console.exe" "C:\...\some.dll" --noresult --noheader --where cat=='cat_name' --framework=net-4.0 --explore

Tests execute successfully when dropping `--explore`

Thanks

**Comment by ChrisMaddock** [comment]
Hi @oryz - Can you try running with `--inprocess` - see if you get a different stack trace?

We could really do with a getting a better stack trace in these situations. :-(

**Comment by ChrisMaddock** [comment]
Unrelated - you should be aware `--explore` doesn't currently work with filters (`--where`) - but @Poimen is working on it, so hopefully it will for the next release! See: https://github.com/nunit/nunit-console/issues/139

**Comment by CharliePoole** [comment]
Looks like a console bug we fixed. What version us this?

**Comment by eberlid** [comment]
The exception stack trace looks the same we encountered in https://github.com/nunit/nunit-console/issues/171

**Comment by oryz** [comment]
Hmm weirdly adding --inprocess causes the command to run correctly (albeit disregarding the --where option as per @ChrisMaddock's comment). See - https://www.screencast.com/t/QJtwqX3k

Nunit version = 3.60

**Comment by jnm2** [comment]
We believe this was fixed by https://github.com/nunit/nunit-console/pull/223 which will be released as part of NUnit.Console 3.7. If you'd like to confirm sooner that the issue is fixed, please try `3.7.0-dev-03641` from the `https://ci.appveyor.com/nuget/nunit-console` NuGet feed.

If it turns out that this does not fix your issue, please open a new issue at https://github.com/nunit/nunit-console/issues.
